### PR TITLE
SONAR-8857 fix quality flaws

### PR DESCRIPTION
**Pull request by bartfastiel**
:ocean: This PR is based on #1816, which should be reviewed first.

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SearchAction.java [hunk 1]
```diff
@@ -125,8 +128,10 @@ SearchWsResponse doHandle(SearchWsRequest request) {
...
   private SearchData load(SearchWsRequest request) {
     try (DbSession dbSession = dbClient.openSession(false)) {
+      OrganizationDto organization = wsSupport.getOrganizationByKey(dbSession, request.getOrganizationKey());
       return new SearchData()
-        .setProfiles(dataLoader.findProfiles(dbSession, request))
+        .setOrganization(organization)
+        .setProfiles(dataLoader.findProfiles(dbSession, request, organization))
         .setActiveRuleCountByProfileKey(activeRuleIndex.countAllByQualityProfileKey())
         .setActiveDeprecatedRuleCountByProfileKey(activeRuleIndex.countAllDeprecatedByQualityProfileKey())
         .setProjectCountByProfileKey(dbClient.qualityProfileDao().countProjectsByProfileKey(dbSession));
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SearchAction.java]: MINOR Remove this use of "countProjectsByProfileKey"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java [hunk 2]
```diff
@@ -176,10 +176,18 @@ public QualityProfileDto selectByNameAndLanguage(OrganizationDto organization, S
     return executeLargeInputs(languageKeys, input -> mapper(session).selectByNameAndLanguages(organization.getUuid(), name, input));
   }
 
+  /**
+   * @deprecated provide organization
+   */
+  @Deprecated
   public List<ComponentDto> selectProjects(String profileName, String language, DbSession session) {
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java [hunk 3]
```diff
@@ -176,10 +176,18 @@ public QualityProfileDto selectByNameAndLanguage(OrganizationDto organization, S
...
     return mapper(session).selectProjects(profileName, language);
   }
 
+  /**
+   * @deprecated provide organization
+   */
+  @Deprecated
   public Map<String, Long> countProjectsByProfileKey(DbSession dbSession) {
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 10 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 6 info

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L122](https://github.com/SonarSource/sonarqube/blob/a8972c69966f11643c5024e5dd1e5b67bbc43c04/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L122): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L133](https://github.com/SonarSource/sonarqube/blob/a8972c69966f11643c5024e5dd1e5b67bbc43c04/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L133): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [CreateAction.java#L64](https://github.com/SonarSource/sonarqube/blob/a8972c69966f11643c5024e5dd1e5b67bbc43c04/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/CreateAction.java#L64): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)
1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/a8972c69966f11643c5024e5dd1e5b67bbc43c04/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/a8972c69966f11643c5024e5dd1e5b67bbc43c04/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QProfileWsSupport.java#L79](https://github.com/SonarSource/sonarqube/blob/a8972c69966f11643c5024e5dd1e5b67bbc43c04/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L79): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QProfileWsSupport.java#L98](https://github.com/SonarSource/sonarqube/blob/a8972c69966f11643c5024e5dd1e5b67bbc43c04/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L98): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 11 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 6 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CachingRuleActivatorContextFactory.java#L74](https://github.com/SonarSource/sonarqube/blob/f5afd40178f545bb525a6046ddb0f0ff2b4effa6/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/CachingRuleActivatorContextFactory.java#L74): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L122](https://github.com/SonarSource/sonarqube/blob/f5afd40178f545bb525a6046ddb0f0ff2b4effa6/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L122): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L133](https://github.com/SonarSource/sonarqube/blob/f5afd40178f545bb525a6046ddb0f0ff2b4effa6/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L133): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [CreateAction.java#L64](https://github.com/SonarSource/sonarqube/blob/f5afd40178f545bb525a6046ddb0f0ff2b4effa6/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/CreateAction.java#L64): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)
1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/f5afd40178f545bb525a6046ddb0f0ff2b4effa6/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/f5afd40178f545bb525a6046ddb0f0ff2b4effa6/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QProfileWsSupport.java#L79](https://github.com/SonarSource/sonarqube/blob/f5afd40178f545bb525a6046ddb0f0ff2b4effa6/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L79): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QProfileWsSupport.java#L98](https://github.com/SonarSource/sonarqube/blob/f5afd40178f545bb525a6046ddb0f0ff2b4effa6/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L98): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 11 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 6 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CachingRuleActivatorContextFactory.java#L74](https://github.com/SonarSource/sonarqube/blob/59cdea5bf86387cb197abf6134abab01beccb532/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/CachingRuleActivatorContextFactory.java#L74): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L122](https://github.com/SonarSource/sonarqube/blob/59cdea5bf86387cb197abf6134abab01beccb532/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L122): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L133](https://github.com/SonarSource/sonarqube/blob/59cdea5bf86387cb197abf6134abab01beccb532/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L133): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [CreateAction.java#L64](https://github.com/SonarSource/sonarqube/blob/59cdea5bf86387cb197abf6134abab01beccb532/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/CreateAction.java#L64): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)
1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/59cdea5bf86387cb197abf6134abab01beccb532/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/59cdea5bf86387cb197abf6134abab01beccb532/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QProfileWsSupport.java#L79](https://github.com/SonarSource/sonarqube/blob/59cdea5bf86387cb197abf6134abab01beccb532/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L79): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QProfileWsSupport.java#L98](https://github.com/SonarSource/sonarqube/blob/59cdea5bf86387cb197abf6134abab01beccb532/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L98): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileService.java [hunk 4]
```diff
@@ -87,6 +87,7 @@ public BulkChangeResult bulkDeactivate(RuleQuery ruleQuery, String profile) {
   }
 
   private void verifyAdminPermission() {
+    // FIXME check for the permission of the appropriate organization, not the default one
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileService.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)

**Comment by SonarTech** [comment]
SonarQube analysis reported 12 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 6 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CachingRuleActivatorContextFactory.java#L74](https://github.com/SonarSource/sonarqube/blob/1d4ac4b12afb4a2d039e9fcd7214cf92997be466/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/CachingRuleActivatorContextFactory.java#L74): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L122](https://github.com/SonarSource/sonarqube/blob/1d4ac4b12afb4a2d039e9fcd7214cf92997be466/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L122): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L133](https://github.com/SonarSource/sonarqube/blob/1d4ac4b12afb4a2d039e9fcd7214cf92997be466/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L133): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [CreateAction.java#L64](https://github.com/SonarSource/sonarqube/blob/1d4ac4b12afb4a2d039e9fcd7214cf92997be466/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/CreateAction.java#L64): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)
1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/1d4ac4b12afb4a2d039e9fcd7214cf92997be466/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/1d4ac4b12afb4a2d039e9fcd7214cf92997be466/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QProfileWsSupport.java#L79](https://github.com/SonarSource/sonarqube/blob/1d4ac4b12afb4a2d039e9fcd7214cf92997be466/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L79): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QProfileWsSupport.java#L98](https://github.com/SonarSource/sonarqube/blob/1d4ac4b12afb4a2d039e9fcd7214cf92997be466/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L98): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 8 issues
* ![MAJOR][MAJOR] 4 major
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CachingRuleActivatorContextFactory.java#L74](https://github.com/SonarSource/sonarqube/blob/f9a84b4170590121f839cce8d41fa24425fc242c/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/CachingRuleActivatorContextFactory.java#L74): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L122](https://github.com/SonarSource/sonarqube/blob/f9a84b4170590121f839cce8d41fa24425fc242c/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L122): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L133](https://github.com/SonarSource/sonarqube/blob/f9a84b4170590121f839cce8d41fa24425fc242c/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L133): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [CreateAction.java#L64](https://github.com/SonarSource/sonarqube/blob/f9a84b4170590121f839cce8d41fa24425fc242c/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/CreateAction.java#L64): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)
1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/f9a84b4170590121f839cce8d41fa24425fc242c/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/f9a84b4170590121f839cce8d41fa24425fc242c/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java [hunk 5]
```diff
@@ -57,7 +57,7 @@ public void define(NewController controller) {
       .setPost(true)
       .setHandler(this);
 
-    qProfileWsSupport.createOrganizationParam(action)
+    QProfileWsSupport.createOrganizationParam(action)
```
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java]: could/should be merged with previous commit (same title and same change, just at a second location)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java [hunk 6]
```diff
@@ -78,24 +78,28 @@ public void resetLanguage(DbSession dbSession, OrganizationDto organization, Str
...
-          }
-          activations.add(activation);
+          activations.add(getRuleActivation(dbSession, activeRule));
         }
       }
       reset(dbSession, profile, activations);
     }
   }
 
+  private RuleActivation getRuleActivation(DbSession dbSession, ActiveRule activeRule) {
```
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java]: couldn't this method be static?

Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java]: No, it uses `db`.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java (hunk 5 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java]: Done. 🦑

File: server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java [hunk 7]
```diff
@@ -228,8 +228,11 @@ public void deleteAllProjectProfileAssociation(String profileKey, DbSession sess
     return mapper(session).selectProjectAssociations(profileKey, nameQuery);
   }
 
-  private String sqlQueryString(@Nullable String query) {
-    return query == null ? "%" : "%" + query.toUpperCase(Locale.ENGLISH) + "%";
+  private static String sqlQueryString(@Nullable String query) {
```
Comment (sns-seb) [inline server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java]: I agree that the new version is more readable (and you could use a constant for "%"), but how is this a quality flaw?
It's quite a matter of taste
another reason to make this change part of the same commit of all the other "fix quality flaws" commit

Comment (bartfastiel) [inline server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java]: According to SonarLint this method contained 3 issues. I solved them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 8 issues
* ![MAJOR][MAJOR] 4 major
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CachingRuleActivatorContextFactory.java#L74](https://github.com/SonarSource/sonarqube/blob/7e49c088ad05d0918394a38c54f5b8e09a06f8a3/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/CachingRuleActivatorContextFactory.java#L74): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L122](https://github.com/SonarSource/sonarqube/blob/7e49c088ad05d0918394a38c54f5b8e09a06f8a3/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L122): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L133](https://github.com/SonarSource/sonarqube/blob/7e49c088ad05d0918394a38c54f5b8e09a06f8a3/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L133): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [CreateAction.java#L64](https://github.com/SonarSource/sonarqube/blob/7e49c088ad05d0918394a38c54f5b8e09a06f8a3/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/CreateAction.java#L64): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)
1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/7e49c088ad05d0918394a38c54f5b8e09a06f8a3/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/7e49c088ad05d0918394a38c54f5b8e09a06f8a3/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 8 issues
* ![MAJOR][MAJOR] 4 major
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CachingRuleActivatorContextFactory.java#L74](https://github.com/SonarSource/sonarqube/blob/f992eb8e6a658cb8aff97671427625965eab17b3/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/CachingRuleActivatorContextFactory.java#L74): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L122](https://github.com/SonarSource/sonarqube/blob/f992eb8e6a658cb8aff97671427625965eab17b3/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L122): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [QProfileResetImpl.java#L133](https://github.com/SonarSource/sonarqube/blob/f992eb8e6a658cb8aff97671427625965eab17b3/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java#L133): Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [CreateAction.java#L64](https://github.com/SonarSource/sonarqube/blob/f992eb8e6a658cb8aff97671427625965eab17b3/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/CreateAction.java#L64): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)
1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/f992eb8e6a658cb8aff97671427625965eab17b3/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/f992eb8e6a658cb8aff97671427625965eab17b3/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**